File: astenn-core/src/main/csharp/src/Org/Lestr/Astenn/Plugin/IPersistenceDriver.cs
/*
 * Crée par SharpDevelop.
 * Utilisateur: Choupis
 * Date: 30/04/2010
 * Heure: 22:31
 * 
 * Pour changer ce modèle utiliser Outils | Options | Codage | Editer les en-têtes standards.
 */
using System;
using System.Collections.Generic;

namespace Org.Lestr.Astenn.Plugin
{
	
	
	public interface IPersistenceDriver {
	
	
	    void AddPluginInterface(string pluginInterfaceName);
	
	
	    void RemovePluginInterface(string pluginInterfaceName);
	
	
	    bool ExistPluginInterface(string pluginInterfaceName);


        void AddPluginImplementation(string pluginInterfaceName, string pluginImplementationAddress);


        void RemovePluginImplementation(string pluginInterfaceName, string pluginImplementationAddress);


        bool ExistPluginImplementation(string pluginInterfaceName, string pluginImplementationAddress);
	
	
	    IEnumerable<string> GetPluginInterfacesNames();
	
	
	    IEnumerable<string> GetPluginImplementationsAddresses(string pluginInterfaceName);
	
	    
	}// END Interface PersistenceDriver
	
		
}// END Namespace Astenn.Configuration

File: astenn-core/src/main/java/org/lestr/astenn/plugin/IPersistenceDriver.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.lestr.astenn.plugin;


/**
 *
 * @author PIBONNIN
 */
public interface IPersistenceDriver {


    public void addPluginInterface(String pluginInterfaceName);


    public void removePluginInterface(String pluginInterfaceName);


    public boolean existPluginInterface(String pluginInterfaceName);


    public void addPluginImplementation(String pluginInterfaceName, String pluginImplementationAddress);


    public void removePluginImplementation(String pluginInterfaceName, String pluginImplementationAddress);


    public boolean  existPluginImplementation(String pluginInterfaceName, String pluginImplementationAddress);


    public Iterable<String> getPluginInterfacesNames();


    public Iterable<String> getPluginImplementationsAddresses(String pluginInterfaceName);

    
}
